README.adoc

```diff
@@ -77,35 +77,15 @@ Supported config properties include:
 |Yes
 |URL identifying the CockroachDB instance(s) to connect to.  CockroachDB is used for all persistent data.
 
-|`dropshot_external`
```

Comment:
For aforementioned reasons, I'm trying to:

- Take as input an "external IP address"
- Host two dropshot servers, on different ports

Dropshot's configuration expects a SocketAddr, which specifies the IP + port in the configuration file. If I opt to ignore these values (picking 80 and 443 instead, for the port), it felt odd to have a user provide them at all. In lieu of that, I wanted to modify the Nexus configuration to be "precisely the toggles we let a user actually change", no more no less.

Additionally, I found this comment in the Nexus config a little concerning:

```
# Allow larger request bodies (1MiB) to accomodate firewall endpoints (one
# rule is ~500 bytes)
request_body_max_bytes = 1048576
```

I understand why the dropshot config is the way that it is, but if Nexus requires it to be set very particularly to operate correctly, then IMO Nexus should be responsible for setting that config, rather than transparently propagating values which could be inappropriate.

WDYT? This definitely veers into opinion territory.

Reply:
> For aforementioned reasons, I'm trying to:
> 
>     * Take as input an "external IP address"
> 
>     * Host two dropshot servers, on different ports
> 
> 
> Dropshot's configuration expects a SocketAddr, which specifies the IP + port in the configuration file. If I opt to ignore these values (picking 80 and 443 instead, for the port), it felt odd to have a user provide them at all. In lieu of that, I wanted to modify the Nexus configuration to be "precisely the toggles we let a user actually change", no more no less.

Sorry I didn't see the discussion of why we want to take an external IP address and choose two ports rather than accept two different IP/port pairs.  (Did I miss that?)

I'm gathering there's kind of a philosophical difference here.  My approach with stuff like Nexus is to have configuration be somewhat general-purpose so that it can be customized by developers and test harnesses for various ways of debugging and deploying it.  (I also understand the desire to avoid providing self-destruct buttons, and I think we can still do that.)

> Additionally, I found this comment in the Nexus config a little concerning:
> 
> ```
> # Allow larger request bodies (1MiB) to accomodate firewall endpoints (one
> # rule is ~500 bytes)
> request_body_max_bytes = 1048576
> ```
> 
> I understand why the dropshot config is the way that it is, but if Nexus requires it to be set very particularly to operate correctly, then IMO Nexus should be responsible for setting that config, rather than transparently propagating values which could be inappropriate.

The reason I originally made this configurable was so that we could in principle tune it _up_ in an emergency (as opposed to so that someone trying to tune it down aggressively for some reason).  The challenge is knowing with sufficient confidence that the value selected is big enough.  Now, the config file may not really be the best place for tunables like this in the deployed system, but I think it's the closest thing we've got today to a place where somebody could tune this up in an emergency without rebuilding the software.

Reply:
Re-added config - the new implementation allows multiple external configs, so this comment thread can be resolved.